### PR TITLE
Smoking and functional lighters

### PR DESCRIPTION
**Pull request by Macoron**
### Purpose
Implemented cigarettes (smoking tube, cigars) and lighters. Finally you can light cigarette by fancy zippo, open plasma canister and enjoy the show.

### Notes:
Added new class for fire sources like lighters, welders and burning cigarettes. They create hotspot and may lead to plasma fire.
Clothing animation seems to not working yet, so burning cigarettes doesn't play burning animation in hands or mouth.

Smoking doesn't affect players health for now.
I didn't find any smoking vending machines on maps, so it's good time to start adding them.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Fire/FireSource.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/// <summary>
+/// Defines fire sources like candles, lighters or activated welder
+/// Fire source can lead to plasma fire or used to ignite flamable objects
+/// </summary>
+public class FireSource : MonoBehaviour, IServerSpawn
+{
+	[SerializeField]
+	[Tooltip("Does this fire emit flame from start?")]
+	private bool isBurningOnSpawn;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Fire/FireSource.cs]: Could you assign a value or `null` or `default` to any non-public `[SerializeField]` fields? Just helps keep the warnings down. Thanks.

File: UnityProject/Assets/Scripts/Fire/FireSource.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	/// Flame may lead to ignite flamable objects or plasma fire
+	/// </summary>
+	public bool IsBurning { get; set; }
+
+	private void Awake()
+	{
+		pushPull = GetComponent<PushPull>();
+	}
+
+	private void Update()
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Fire/FireSource.cs]: I would suggest using `UpdateManager` to handle updates. That way you can subscribe for updates only when needed - for example when it is the server and `pushPull` is not null, and `IsBurning`.

**Comment by ReiderJack** [comment]
I like how you name booleans! 👍

File: UnityProject/Assets/Scripts/Fire/FireSource.cs (hunk 1 above)
Comment (Macoron) [reply UnityProject/Assets/Scripts/Fire/FireSource.cs]: Should be done now

File: UnityProject/Assets/Scripts/Fire/FireSource.cs (hunk 2 above)
Comment (Macoron) [reply UnityProject/Assets/Scripts/Fire/FireSource.cs]: Yep, implemented that. Didn't touch Cigarette and Lighters update loops, cause they will refactored soon for new sprite system.

**Pull request merged**